### DISCUSSION TITLE
Can models be exported as code? [ANSWERED]

### DISCUSSION THREAD

**Discussion by Aer0spaceEngineered** [ultralytics/hub#16] [Q&A] [answered]
Can models be exported as lines of code?

**Comment by glenn-jocher** [comment, marked as answer]
@TheSuperiorTwin not sure what you mean. Every model is composed of code. All supported export formats are indicated here:

## Formats

YOLOv5 inference is officially supported in 11 formats:

Format                      | `export.py --include`         | Model
:---                        | --:                           | :--
[PyTorch](https://pytorch.org/)                     | -                             | `yolov5s.pt`
[TorchScript](https://pytorch.org/docs/stable/jit.html)                 | `torchscript`                 | `yolov5s.torchscript`
[ONNX](https://onnx.ai/)                        | `onnx`                        | `yolov5s.onnx`
[OpenVINO](https://docs.openvino.ai/latest/index.html)                    | `openvino`                    | `yolov5s_openvino_model/`
[TensorRT](https://developer.nvidia.com/tensorrt)                    | `engine`                      | `yolov5s.engine`
[CoreML](https://github.com/apple/coremltools)                      | `coreml`                      | `yolov5s.mlmodel`
[TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model)       | `saved_model`                 | `yolov5s_saved_model/`
[TensorFlow GraphDef](https://www.tensorflow.org/api_docs/python/tf/Graph)         | `pb`                          | `yolov5s.pb`
[TensorFlow Lite](https://www.tensorflow.org/lite)             | `tflite`                      | `yolov5s.tflite`
[TensorFlow Edge TPU](https://coral.ai/docs/edgetpu/models-intro/)         | `edgetpu`                     | `yolov5s_edgetpu.tflite`
[TensorFlow.js](https://www.tensorflow.org/js)               | `tfjs`                        | `yolov5s_web_model/`


You can visualize all exports with Netron, load them in their respective environments for introspection etc. For example if you download your trained PyTorch model:

```python
model = torch.load('downloaded_model.pt')
print(model['model'])
```

<img width="755" alt="Screenshot 2022-04-13 at 15 54 37" src="https://user-images.githubusercontent.com/26833433/163196383-792424e7-68bc-4776-b1d6-5f1767cc3314.png">

**Reply by Aer0spaceEngineered** [reply]
This is what I meant, sorry for the vague wording. Thank you!

**Reply by glenn-jocher** [reply]
Great! Let us know if you have any other questions or ideas for improving HUB :)
